Compilation failure (mismatched types)

It currently doesn't build on Rust 1.65.0

```
error[E0308]: mismatched types
  --> src/auth/utmpx.rs:23:21
   |
23 |         s.ut_user = ut_user;
   |         ---------   ^^^^^^^ expected `u8`, found `i8`
   |         |
   |         expected due to the type of this binding
   |
   = note: expected array `[u8; 32]`
              found array `[i8; 32]`

error[E0308]: mismatched types
  --> src/auth/utmpx.rs:36:21
   |
36 |         s.ut_line = ut_line;
   |         ---------   ^^^^^^^ expected `u8`, found `i8`
   |         |
   |         expected due to the type of this binding
   |
   = note: expected array `[u8; 32]`
              found array `[i8; 32]`

error[E0308]: mismatched types
  --> src/auth/utmpx.rs:40:19
   |
40 |         s.ut_id = ut_id;
   |         -------   ^^^^^ expected `u8`, found `i8`
   |         |
   |         expected due to the type of this binding
   |
   = note: expected array `[u8; 4]`
              found array `[i8; 4]`
```

--- a/src/auth/utmpx.rs
+++ b/src/auth/utmpx.rs
@@ -1,6 +1,6 @@
 use std::time::SystemTime;
 
-use libc::{utmpx, DEAD_PROCESS, USER_PROCESS};
+use libc::{c_char, utmpx};
 use log::{error, info};
 
 pub struct UtmpxSession(utmpx);
@@ -8,36 +8,45 @@ pub struct UtmpxSession(utmpx);
 pub fn add_utmpx_entry(username: &str, tty: u8, pid: u32) -> UtmpxSession {
     info!("Adding UTMPX record");
 
+    // Check the MAN page for utmp for more information
+    // `man utmp`
+    //
     // https://man7.org/linux/man-pages/man0/utmpx.h.0p.html
     // https://github.com/fairyglade/ly/blob/master/src/login.c
     let entry = {
         let mut s: utmpx = unsafe { std::mem::zeroed() };
 
-        s.ut_type = USER_PROCESS;
+        // ut_line    --- Device name of tty - "/dev/"
+        // ut_id      --- Terminal name suffix
+        // ut_user    --- Username
+        // ut_host    --- Hostname for remote login, or kernel version for run-level messages
+        // ut_exit    --- Exit status of a process marked as DEAD_PROCESS; not used by Linux init(1)
+        // ut_session --- Session ID (getsid(2)) used for windowing
+        // ut_tv {    --- Time entry was made
+        //     tv_sec     --- Seconds
+        //     tv_usec    --- Microseconds
+        // }       
+        // ut_addr_v6 --- Internet address of remote
+
+        s.ut_type = libc::USER_PROCESS;
         s.ut_pid = pid as libc::pid_t;
 
-        let mut ut_user = [0; 32];
         for (i, b) in username.as_bytes().iter().take(32).enumerate() {
-            ut_user[i] = *b as i8;
+            s.ut_user[i] = *b as c_char;
         }
-        s.ut_user = ut_user;
 
         if tty > 12 {
             error!("Invalid TTY");
             std::process::exit(1);
         }
-        let tty_c_char = (b'0' + tty) as i8;
+        let tty_c_char = (b'0' + tty) as c_char;
 
-        let mut ut_line = [0; 32];
-        ut_line[0] = b't' as i8;
-        ut_line[1] = b't' as i8;
-        ut_line[2] = b'y' as i8;
-        ut_line[3] = tty_c_char;
-        s.ut_line = ut_line;
+        s.ut_line[0] = b't' as c_char;
+        s.ut_line[1] = b't' as c_char;
+        s.ut_line[2] = b'y' as c_char;
+        s.ut_line[3] = tty_c_char;
 
-        let mut ut_id = [0; 4];
-        ut_id[0] = tty_c_char;
-        s.ut_id = ut_id;
+        s.ut_id[0] = tty_c_char;
 
         let epoch_duration = SystemTime::now()
             .duration_since(SystemTime::UNIX_EPOCH)
@@ -59,8 +68,10 @@ pub fn add_utmpx_entry(username: &str, tty: u8, pid: u32) -> UtmpxSession {
         s
     };
 
-    unsafe { libc::setutxent() };
-    unsafe { libc::pututxline(&entry as *const utmpx) };
+    unsafe {
+        libc::setutxent();
+        libc::pututxline(&entry as *const utmpx);
+    };
 
     info!("Added UTMPX record");
 
@@ -73,10 +84,10 @@ impl Drop for UtmpxSession {
 
         info!("Removing UTMPX record");
 
-        entry.ut_type = DEAD_PROCESS;
+        entry.ut_type = libc::DEAD_PROCESS;
 
-        entry.ut_line = [0; 32];
-        entry.ut_user = [0; 32];
+        entry.ut_line = <[c_char; 32]>::default();
+        entry.ut_user = <[c_char; 32]>::default();
 
         entry.ut_tv.tv_usec = 0;
         entry.ut_tv.tv_sec = 0;

--- a/src/auth/utmpx.rs
+++ b/src/auth/utmpx.rs
@@ -25,7 +25,7 @@ pub fn add_utmpx_entry(username: &str, tty: u8, pid: u32) -> UtmpxSession {
         // ut_tv {    --- Time entry was made
         //     tv_sec     --- Seconds
         //     tv_usec    --- Microseconds
-        // }       
+        // }
         // ut_addr_v6 --- Internet address of remote
 
         s.ut_type = libc::USER_PROCESS;

